src/transformers/modeling_utils.py

```diff
@@ -2356,6 +2366,28 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         # Set model in evaluation mode to deactivate DropOut modules by default
         model.eval()
 
+        # If it is a model with generation capabilities, attempt to set the generation config to an existing
+        # `generation_config.json` file.
+        if model.can_generate():
+            try:
+                generation_config = GenerationConfig.from_pretrained(
+                    pretrained_model_name_or_path,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    resume_download=resume_download,
+                    proxies=proxies,
+                    local_files_only=local_files_only,
+                    use_auth_token=use_auth_token,
+                    revision=revision,
+                    subfolder=subfolder,
+                    _from_auto=from_auto_class,
+                    _from_pipeline=from_pipeline,
+                    **kwargs,
+                )
+                model.generation_config = generation_config
+            except EnvironmentError:
+                pass
```

Comment:
Should `model.generation_config` be set to `None` here?

Reply:
Actually shouldn't we always create a generation_config even if no generation_config is present yet? Or add a parameter to `from_pretrained(..., create_if_not_exist=True)` ? 

Every generation model that is created after this PR should have a generation_config.json no? So I'd say we always create it automatically from `config.json` if `create_if_not_exist=True`.

Reply:
If I understand the code correctly it means that no generation config will be automatically created going forward now => so I don't see how we can make the generation config the default case this way.

Reply:
I see we create it here, but then don't set it. Shouldn't we do this line: https://github.com/huggingface/transformers/pull/20388/files#r1035839081 also in the case above?

Or maybe instead of adding a `create_if_not_exist` arg we could do:
```suggestion
                logger.info("Creating generation config from model config since no generation config was found in .... Since, v4.25.0 generation configs are becoming the default way of controlling the generate method. For more information you can see: ...")
                generation_config = GenerationConfig.from_model_config(self.config)
           except EnvironmentError:
```

Reply:
We could also save the `generation_config` in `from_pretrained` so that we get it automatically on the repo when pushing to the Hub, maybe?